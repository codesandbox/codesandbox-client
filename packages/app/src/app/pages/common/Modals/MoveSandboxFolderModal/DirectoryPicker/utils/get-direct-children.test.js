import getDirectChildren from './get-direct-children';

describe('get-direct-children', () => {
  const defaultItems = [
    { path: '/test' },
    { path: '/test/test2' },
    { path: '/test2' },
  ];

  it('works with root path', () => {
    const root = '/';

    expect(getDirectChildren(root, defaultItems)).toMatchSnapshot();
  });

  it('works with sub paths', () => {
    const root = '/test';

    expect(getDirectChildren(root, defaultItems)).toMatchSnapshot();
  });

  it('works with deeper paths', () => {
    const items = [
      { path: '/test/test2/test3' },
      { path: '/test/test2' },
      { path: '/test/test3' },
      { path: '/test/test2/test4' },
      { path: '/test/test2/test3' },
    ];

    expect(getDirectChildren('/test', items)).toMatchSnapshot();
  });
});
