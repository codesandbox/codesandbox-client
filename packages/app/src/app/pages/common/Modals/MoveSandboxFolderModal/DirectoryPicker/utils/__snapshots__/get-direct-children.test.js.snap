// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-direct-children works with deeper paths 1`] = `
Set {
  "test2",
  "test3",
}
`;

exports[`get-direct-children works with root path 1`] = `
Set {
  "test",
  "test2",
}
`;

exports[`get-direct-children works with sub paths 1`] = `
Set {
  "test2",
}
`;
